### PR TITLE
Change the userErrors name

### PR DESCRIPTION
**Pull request by sabelard-bold**
- Make the userErrors field name to be the name of the mutation instead

### CODE REVIEW

File: generate/generate.go [hunk 1]
```diff
@@ -116,18 +116,19 @@ func (g *Generator) WriteFunctions() error {
...
 				if !ok {
 					return fmt.Errorf("could not find type for mutation user error: %s", field.TypeName())
 				}
-
 				break
 			}
 		}
 
-		mtm := newMutationTemplateModel(f, ret, g.schema.Mutation.Type, ue)
+		mtm := newMutationTemplateModel(f, ret, g.schema.Mutation.Type, ue, errorField)
```
Comment (Wryte) [inline generate/generate.go]: Seems like we should be able to change the params of this method if we make the graphql structs more capable. I could see it being something like

```golang
newMutationTemplateModel(g.schema, f)
```

And then inside we do the work to get what we need out of that mutation Field.

Comment (sabelard-bold) [reply generate/generate.go]: Done, I also tried to move as much code as possible as struct funtion when it make sense. Let me know what you think. I tested it does generate the same code as previously :)

File: graphql/type.go [hunk 2]
```diff
@@ -33,3 +33,14 @@ const (
 func (t *Type) IsInputObject() bool {
 	return t.Kind == TypeKindInputObject
 }
+
+// GetErrorField return the error field if there is one
+func (t *Type) GetErrorField() *Field {
```
Comment (Wryte) [inline graphql/type.go]: nit: I think go conventions say this should just be `ErrorField()`

Comment (sabelard-bold) [reply graphql/type.go]: Fixed

**Pull request merged**